packages/serverpod/lib/src/server/endpoint_dispatch.dart

```diff
@@ -336,13 +359,102 @@ class MethodStreamCallContext {
   /// The input streams to pass to the method.
   final List<StreamParameterDescription> inputStreams;
 
+  OnRevokedAuthenticationCallback? _onRevokedAuthentication;
+
+  MessageCentralListenerCallback? _revokedAuthenticationCallback;
+
+  final Session _session;
+
+  AuthenticationInfo? _authenticationInfo;
+
   /// Creates a new [MethodStreamCallContext].
   MethodStreamCallContext({
     required this.method,
     required this.arguments,
     required this.inputStreams,
     required this.endpoint,
-  });
+    required Session session,
+  }) : _session = session;
+
+  /// Initializes the context.
+  Future<void> init() async {
+    var authenticationIsRequired =
+        endpoint.requireLogin || endpoint.requiredScopes.isNotEmpty;
+
+    if (authenticationIsRequired) {
+      var authenticationInfo = await _session.authenticated;
+      if (authenticationInfo == null) {
+        throw NotAuthorizedException(
```

Comment:
Suggestion: This should be a state error since we have already validated that the user it authorized to make this call.